# For a direct USB cable connection, the port name is typically
# /dev/ttyACM# where is # is a number such as 0, 1, 2, etc
# For a wireless connection like XBee, the port is typically
# /dev/ttyUSB# where # is a number such as 0, 1, 2, etc.

port: /dev/ttyAMA0
baud: 57600
timeout: 0.1

rate: 50
sensorstate_rate: 10

use_base_controller: True
base_controller_rate: 10

# For a robot that uses base_footprint, change base_frame to base_footprint
base_frame: base_link

# === Robot drivetrain parameters
wheel_diameter: 0.0625
wheel_track: 0.2794
encoder_resolution: 1856 # http://www.robotshop.com/en/pololu-12v-29-1-gear-motor-encoder.html
gear_reduction: 1
motors_reversed: False

# === PID parameters
Kp: 10
Kd: 12
Ki: 0
Ko: 50
accel_limit: 1.0

# === Sensor definitions.  Examples only - edit for your robot.
#     Sensor type can be one of the follow (case sensitive!):
#   * Ping
#   * GP2D12
#   * Analog
#   * Digital
#   * PololuMotorCurrent
#   * PhidgetsVoltage
#   * PhidgetsCurrent (20 Amp, DC)



sensors: {
  #motor_current_left:   {pin: 0, type: PololuMotorCurrent, rate: 5},
  #motor_current_right:  {pin: 1, type: PololuMotorCurrent, rate: 5},
  #ir_front_center:      {pin: 2, type: GP2D12, rate: 10},
  #sonar_front_center:   {pin: 5, type: Ping, rate: 10},
  arduino_led:           {pin: 13, type: Digital, rate: 5, direction: output}
}


use_joint_controller: True
joint_controller_rate: 10

joints: {
    right_armroot_to_armlink1:  {id: 0},
    right_armlink1_to_armlink2: {id: 1},
    right_armlink2_to_armlink3: {id: 2},
    right_armlink3_to_armlink4: {id: 3},
    right_armlink4_to_armlink5: {id: 4},
    right_gripper_joint:        {id: 5},
    left_armroot_to_armlink1:   {id: 6},
    left_armlink1_to_armlink2:  {id: 7},
    left_armlink2_to_armlink3:  {id: 8},
    left_armlink3_to_armlink4:  {id: 9},
    left_armlink4_to_armlink5:  {id: 10},
    left_gripper_joint:         {id: 11}
}
